eclass/blas.eclass

```diff
@@ -0,0 +1,241 @@
+# @ECLASS: blas.eclass
+# @MAINTAINER:
+# schmidom@student.ethz.ch
+# @BLURB: Selects a proper BLAS implementation to build the package
+# @DESCRIPTION:
+# Here a BLAS-implementation is chosen from the cutset of available,
+# compatible and implementations in the USE-flags
+# The implementation is then enforced upon every dependency of the package
+
+# @ECLASS-VARIABLE: BLAS_COMPAT
+# @REQUIRED
+# @DESCRIPTION:
+# This variable contains a list of BLAS implementations the package
+# supports. It has to be an array. Either it or BLAS_COMPAT_ALL must be
+# set before the `inherit' call. BLAS_COMPAT_ALL overrides BLAS_COMPAT.
+#
+# Example:
+# @CODE
+# BLAS_COMPAT=( refblas openblas gotoblas mkl )
+# @CODE
+
+# @ECLASS-VARIABLE: BLAS_COMPAT_ALL
+# @REQUIRED
+# @DESCRIPTION:
+# This variable marks that the package is compatible with all (standard)
+# BLAS implementations. Set it to a non-empty value to make your package
+# compatible with any BLAS version. Either it or BLAS_COMPAT must be set
+# before the `inherit' call. BLAS_COMPAT_ALL overrides BLAS_COMPAT.
+#
+# Example:
+# @CODE
+# BLAS_COMPAT_ALL=1
+# @CODE
+
+# @ECLASS-VARIABLE: BLAS_USEDEP
+# @DESCRIPTION:
+# This is an eclass-generated USE-dependency string which has to be used
+# to propagate the chosen BLAS-implementation down the dependency graph
+#
+# Example use:
+# @CODE
+# RDEPEND="sci-libs/foo[${BLAS_USEDEP}]"
+# @CODE
+#
+# Example value:
+# @CODE
+# refblas?,openblas?
+# @CODE
+
+
+# @ECLASS-VARIABLE: BLAS_CONDITIONAL_FLAG
+# @DESCRIPTION:
+# This variable contains the USE-flag that selects whether the package
+# should depend on BLAS or not.
+# If non-empty, it gets prepended to REQUIRED_USE, DEPEND and RDEPEND.
+# If it is an array, we will depend on BLAS when at least one of the 
+# USE-flags is activated (logical OR)
+#
+# If, for example, set to
+# @CODE
+# BLAS_CONDITIONAL_FLAG="blas"
+# @CODE
+#
+# Then the eclasses REQUIRED_USE and DEPENDs look like this:
+# @CODE
+# DEPEND="blas? ( blas_refblas? ( sci-libs/blas-reference ))"
+# RDEPEND="${DEPEND}"
+# REQUIRED_USE="blas? ( ^^ ( blas_refblas blas_openblas ) )"
+# @CODE
+
+# @ECLASS-VARIABLE: BLAS_USE_CBLAS
```

Comment:
Do you always use CBLAS along with BLAS, or is there a use case for CBLAS alone?

Reply:
I am not sure what you mean here. But usually you one or the other. BLAS is a fortran interface and CBLAS is a C interface. For historical reason the fortran interface is the one most used and some implementation like openblas combine the two in one fat library.